src/sharp_plugin.c

```diff
@@ -95,8 +95,12 @@ int ncclSharpAllGather(void *context, void *buf, int len) {
   struct ncclSharpCollComm* cComm = (struct ncclSharpCollComm*)context;
 
   void* rMhandle, *sMhandle;
-  NCCLCHECK(NCCL_PLUGIN_SYMBOL.regMr(cComm->recvComm, buf, cComm->nranks*len, NCCL_PTR_HOST, &rMhandle));
-  NCCLCHECK(NCCL_PLUGIN_SYMBOL.regMr(cComm->sendComm, buf, cComm->nranks*len, NCCL_PTR_HOST, &sMhandle));
+
+  if (strcmp(NCCL_PLUGIN_SYMBOL.name, "UCX") != 0) {
```

Comment:
instead of strcmp, can we introduce variable with enum to indicate which Ptp ( NCCL_P2P_IB, NCCL_P2P_UCX)